app/gui2/src/components/TextFormattingSelector.vue

```diff
@@ -13,18 +13,40 @@ const emit = defineEmits<{
 const textFormatterSelected = ref(TextFormatOptions.Partial)
 watch(textFormatterSelected, (selected) => emit('changeFormat', selected))
 
+const buttonClass = computed(() => {
+  return {
+    full: isFormatOptionSelected(TextFormatOptions.On),
+    partial: isFormatOptionSelected(TextFormatOptions.Partial),
+    strikethrough: isFormatOptionSelected(TextFormatOptions.Off),
+  }
+})
+
+const isFormatOptionSelected = (option: TextFormatOptions): boolean =>
+  option === textFormatterSelected.value
+
 const open = ref(false)
+const toggleOpen = () => {
+  open.value = !open.value
+}
+
+const changeFormat = (option: TextFormatOptions) => {
+  textFormatterSelected.value = option
+  toggleOpen()
+}
 </script>
 
 <template>
   <DropdownMenu v-model:open="open" class="TextFormattingSelector">
```

Comment:
In a selection menu, we usually make each button toggleable. It's OK as is for now; let's get this into the release. There is a `SelectionDropdown` component that could help; it implements this whole pattern of multiple toggleable options, with the current selection shown as the menu button.